Sources/Theme.php

```diff
@@ -1290,18 +1313,12 @@ public static function template_header(): void
 
 		header('content-type: text/' . (isset($_REQUEST['xml']) ? 'xml' : 'html') . '; charset=' . (empty(Utils::$context['character_set']) ? 'ISO-8859-1' : Utils::$context['character_set']));
 
-		// We need to splice this in after the body layer, or after the main layer for older stuff.
+		// We need to splice this in after the body layer.
 		if (Utils::$context['in_maintenance'] && User::$me->is_admin) {
 			$position = array_search('body', Utils::$context['template_layers']);
 
-			if ($position === false) {
-				$position = array_search('main', Utils::$context['template_layers']);
-			}
-
 			if ($position !== false) {
-				$before = array_slice(Utils::$context['template_layers'], 0, $position + 1);
-				$after = array_slice(Utils::$context['template_layers'], $position + 1);
-				Utils::$context['template_layers'] = array_merge($before, ['maint_warning'], $after);
+				array_splice(Utils::$context['template_layers'], $position + 1, 0, ['maint_warning']);
```

Comment:
By the way, what is `maint_warning`? Maybe `main_warning`, `main`, or `warning`?

Reply:
Maintenance warning.